doc/events.rst

```diff
@@ -0,0 +1,98 @@
+Kansha events
+=============
+
+Definition
+----------
+
+Events are tuples in the form `(action, source, data)` where:
+
+* `action` is a string;
+* `source` is a list of business objects, each component relaying the event up must append itself to it; Is is a kind of path in reverse order;
+* `data` is data specific to the action.
+
+Events are returned as Nagare `answer` parameters by components.
+
+List of events
+--------------
+
+Actions end with verbs in base form most of the time. We don't mutate actions to conjugate them and mark they are done; that would not bring any improvement anyway.
+A business object who is concerned by an action knows what to do with it, othewise it just "bubbles" it up.
+
+The most significant actions may be bubbled after they are done too, to trigger some behavior in upper levels (e.g. history logging, indexing…). The action code is left untouched.
+
+Core
+""""
+
+card_create
+    Card creation. The new card is the first source. No data.
+
+card_delete
+    Card deletion. The first source is None instead of the card once the card is deleted and the event relayed. The data is the tuple (id, title) of the deleted card.
+    Beware: the id is **not** the DataCard id but the application card id in the form "card_X".
+
+card_archive
+    Card is archived. No data.
+
+card_edit
+    Toggle card edition on. The data is the card component. Not bubbled after execution.
+
+card_close
+    Card edition is toggled off. Bubbled after execution. The first source is the card. Data is None.
+
+card_move
+    Card has moved between columns. The first source is the card, then the destination column. Data is the origin column.
+
+column_delete
+    Column deletion. The first source is the column to delete, when emitted, then None when it is actually deleted and the event is relayed.
+
+popin_close
+    Emitted by Popin upon close.  The data is the popin Component and the first source is the business object embedded in the Popin.
+
+
+Extensions
+""""""""""
+
+card_weight
```

Comment:
I don't see why such events are available since only the extensions know about them. So no other component can handle them.